plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -799,7 +799,15 @@ public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session,
                         icebergTable.schema().findType(field.sourceId())))
                 .toArray(Type[]::new);
 
-        AppendFiles appendFiles = transaction.newAppend();
+        AppendFiles appendFiles;
+        if (session.getProperty("merge_manifests_on_write", Boolean.class)) {
```

Comment:
Add a utility method `isMergeManifestsOnWrite` to `IcebergSessionProperties` and use that to access the property value, like we do for the others. See `isOrcBloomFiltersEnabled` for an example.